go/vt/vtgate/planbuilder/sequential.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package planbuilder
+
+import (
+	"vitess.io/vitess/go/vt/vtgate/engine"
+)
+
+type sequential struct {
+	sources []logicalPlan
+}
+
+var _ logicalPlan = (*sequential)(nil)
+
+// Primitive implements the logicalPlan interface
+func (s *sequential) Primitive() engine.Primitive {
+	var sources []engine.Primitive
+	for _, source := range s.sources {
+		prim := source.Primitive()
+		switch dml := prim.(type) {
+		case *engine.Update:
+			dml.PreventAutoCommit = true
```

Comment:
nitpick: feels a little weird doing this here. I think this belongs in the operators and or in the operator_transformer